syntax = "proto3";

package api;

import "google/api/annotations.proto";

service ProdService {
    rpc GetVendorProdTypes(ClientRequestType) returns (ClientResponseType) {
        option (google.api.http) = {
            post: "/v1/api/prodtypes"
            body: "*"
        };
    };
    rpc GetVendorProds(ClientRequestProds) returns (stream ClientResponseProds) {
        option (google.api.http) = {
            post: "/v1/api/prods"
            body: "*"
        };
    };
}

message ClientRequestType {
    string vendor = 1;
}

message ClientResponseType {
    string productType = 1;
}

message ClientRequestProds {
    string vendor = 1;
    string productType = 2;
}

message ClientResponseProds {
    ProdsPrep product = 1;
}

message ProdsPrep {
    string title = 1;
    string url = 2;
    string shortUrl = 3;

}

service StorageService {
    rpc GetProdsDetail(StorageRequest) returns (StorageResponse);
}

message StorageRequest {
    string vendor = 1;
    string productType = 2;
}

message StorageResponse {
    repeated Product prodDetail = 1;
}

message Product {
    string title = 1;
    string url = 2;
}